gatsby-node.js

```diff
@@ -9,59 +9,181 @@ const report = require('gatsby-cli/lib/reporter');
  */
 const identity = obj => obj;
 
+/**
+ * Fetches all records for the current index from Algolia
+ *
+ * @param {AlgoliaIndex} index eg. client.initIndex('your_index_name');
+ * @param {Array<String>} attributesToRetrieve eg. ['modified', 'slug']
+ */
+
+function fetchAlgoliaObjects(index, attributesToRetrieve) {
+  return new Promise((resolve, reject) => {
+    const browser = index.browseAll('', { attributesToRetrieve: ['modified'] });
+    const hits = {};
```

Comment:
a big index can definitely make this go out of memory; maybe a temporary file here that can be written & read as a stream (although that clearly complicates things)

Reply:
Since we are only getting 1-3 fields per record it won't be that big of an issue unless it is in the hundreds of thousands of records.

Reply:
let's leave it like this indeed; we can iterate over that later if it's causing issues